src/dso_api/dynamic_api/remote/clients.py

```diff
@@ -70,7 +72,14 @@ def call(self, request, path="", query_params={}) -> Union[dict, list]:  # noqa:
             raise ServiceUnavailable(str(e)) from e
 
         level = logging.ERROR if response.status >= 400 else logging.INFO
-        logger.log(level, "Proxy call to %s, status %s: %s", url, response.status, response.reason)
+        logger.log(
+            level,
+            "Proxy call to %s, status %s: %s, took: %.3fs",
+            url,
+            response.status,
+            response.reason,
+            (time.perf_counter_ns() - t0) * 1e-9,
+        )
```

Comment:
This way you'll only be logging successful calls. I take it time spend making unsuccessful calls is equally valuable?

Reply:
I'm hoping `urllib3.exceptions.TimeoutError` has the timeout in the error message, though TBH, I didn't check.